ldms/src/sampler/examples/test_sampler/test_sampler.c

```diff
@@ -1020,26 +1020,40 @@ static int __init_set(struct test_sampler_set *ts_set)
 		}
 	}
 
-	mid = ldms_metric_by_name(ts_set->set, "component_id");
-	if (mid >= 0) {
-		v.v_u64 = ts_set->compid;
-		if (ldms_metric_is_array(ts_set->set, mid)) {
-			uint32_t count = ldms_metric_array_get_len(ts_set->set, mid);
-			ldms_metric_array_set(ts_set->set, mid, &v, 0, count);
+	if (ts_set->compid) {
+		mid = ldms_metric_by_name(ts_set->set, "component_id");
+		if (mid >= 0) {
+			v.v_u64 = ts_set->compid;
+			if (ldms_metric_is_array(ts_set->set, mid)) {
+				uint32_t count = ldms_metric_array_get_len(ts_set->set, mid);
+				for (j = 0; j < count; j++)
+					ldms_metric_array_set_u64(ts_set->set, mid, j, ts_set->compid);
+			} else {
+				ldms_metric_set(ts_set->set, mid, &v);
+			}
 		} else {
-			ldms_metric_set(ts_set->set, mid, &v);
```

Comment:
@nichamon, what is v set to when the mid < 0? It seems un-initialized to me.

Reply:
If `mid < 0`, `v` isn't set and isn't used. If `mid < 0`, an error message is logged at line 1035.

Reply:
Should we set the mid to some value like -1? Or just leave it uninitialized. Also it looks like this patch needs to be rebased?

Reply:
I think git diff makes this confusing. I actually deleted lines 1023 - 130 in red and replaced them with lines 1023 - 1038 in green. `ldms_metric_set()`  is called only when `mid >= 0` and `v` is set to a value at line 1026. We don't have to assign values to `mid` and `v`. Did I miss anything?

I just rebased the patch.